modules/cli/src/main/scala/scala/cli/internal/CliLogger.scala

```diff
@@ -85,7 +85,11 @@ class CliLogger(
 
         for (textEdit <- textEditOpt) {
           val bTextEdit = TextEdit(range, textEdit.newText)
-          diag.setData(bTextEdit.toJsonTree())
+          val workspaceEdit = WorkspaceEdit(changes =
+            Map(f.path.getOrElse(os.Path("")).toNIO.toUri().toString() -> Array(bTextEdit))
```

Comment:
I was also a bit unsure of what to do here. `f.path` is `Either[String, Path]`. I'm assuming this is due to diagnostics potentially not being tied to a file URI, but instead to a target? What's an example of this being left here? We can't leave this as is since `""` is invalid here, so it's just a placeholder until I better understand what the left is used for here. If it's truly meant to representation maybe a target, should we instead detect left and then just not even attempt to add `data` since the `WorkspaceEdit` will need a valid URI?

Reply:
If I'm not mistaken, the potential `Left` value is reserved for virtual inputs (command line snippets, piped sources, etc) which wouldn't have a conventional file path and instead have a String indicating the source here: https://github.com/VirtusLab/scala-cli/blob/c878f742d2c0f0b62579f7283c1b88c9ecb4b124/modules/build/src/main/scala/scala/build/input/Element.scala#L19 
example `Left` values would be `"<stdin>-scala-file"` or  `"<snippet>-java-snippet2"`.
https://github.com/VirtusLab/scala-cli/blob/c878f742d2c0f0b62579f7283c1b88c9ecb4b124/modules/build/src/main/scala/scala/build/input/Inputs.scala#L240

Reply:
Ah so that should really never be the URI in a diagnostic since it wouldn't actually be a valid file system URI right? If so, I'll just add logic that for `Left` you can't have a valid URI, therefore no `data`. If right, then you can.

Reply:
Yup, I believe you're correct